File: OPT/TC3AutoDual/T3DualAuto/ADLINK/PCIS-DASK/Samples/.NET/C#/6208/D6208_AO_OnePoint/D6208_AO_OnePoint/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace D6208_AO_OnePoint
{
    public partial class Form1 : Form
    {
        public short m_dev;
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            m_dev = DASK.Register_Card(DASK.PCI_6208V, 0);
            if (m_dev < 0)
            {
                MessageBox.Show("Register_Card error!");
            }
        }

        private void Form1_FormClosing(object sender, FormClosingEventArgs e)
        {
            short ret;
            if (m_dev >= 0)
            {
                ret = DASK.Release_Card((ushort)m_dev);
            }
        }

        private void button1_Click(object sender, EventArgs e)
        {
            short ret;
            double out_value;
            if (Double.TryParse(textBox1.Text, out out_value))
            {
                ret = DASK.AO_VWriteChannel((ushort)m_dev, 0, out_value);
                if (ret < 0)
                {
                    MessageBox.Show("AO_VWriteChannel error!");
                }
            }
            else
            {
                MessageBox.Show("Input error!");
            }
        }
    }
}

File: OPT/TC3AutoDual/T3DualAuto/ADLINK/PCIS-DASK/Samples/.NET/VB.NET/6208/D6208_AO_OnePoint/D6208_AO_OnePoint/Form1.vb
Public Class Form1
    Dim m_dev As Short

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        m_dev = Register_Card(PCI_6208V, 0)
        If (m_dev < 0) Then
            MessageBox.Show("PCI_6208V error!")
        End If
    End Sub

    Private Sub Form1_FormClosing(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles MyBase.FormClosing
        Dim ret As Short
        If (m_dev >= 0) Then
            ret = DASK.Release_Card(m_dev)
        End If
    End Sub

    Private Sub button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button1.Click
        Dim err As Short
        Dim out_value As Double
        If (Double.TryParse(textBox1.Text, out_value)) Then
            err = DASK.AO_VWriteChannel(m_dev, 0, out_value)
            If (err < 0) Then
                MessageBox.Show("AO_VWriteChannel error!")
            End If
        Else
            MessageBox.Show("Input error!")
        End If
    End Sub
End Class
